llms/mlx_lm/models/cohere.py

```diff
@@ -0,0 +1,200 @@
+from dataclasses import dataclass
+from typing import Optional, Tuple
+from functools import partial
+
+import mlx.core as mx
+import mlx.nn as nn
+
+from .base import BaseModelArgs
+
+
+@dataclass
+class ModelArgs(BaseModelArgs):
+    model_type: str
+    hidden_size: int = 8192
+    num_hidden_layers: int = 40
+    intermediate_size: int = 22528
+    num_attention_heads: int = 64
+    num_key_value_heads: int = 64
+    rope_theta: float = 8000000.0
+    vocab_size: int = 256000
+    tie_word_embeddings: bool = True
+    layer_norm_eps: float = 1e-05
+    logit_scale: float = 0.0625
+    attention_bias: bool = False
+
+
+@partial(mx.compile, shapeless=True)
+def ln_norm(x, eps, weight=None, bias=None):
+    t = x.dtype
+    x = x.astype(mx.float32)
+    means = mx.mean(x, axis=-1, keepdims=True)
+    var = mx.var(x, axis=-1, keepdims=True)
+    x = (x - means) * mx.rsqrt(var + eps)
+    x = x.astype(t)
+    h = weight.astype(mx.float32) * x
+    if bias is not None:
+        h = h + bias.astype(mx.float32)
+    return h
```

Comment:
Why not use the LayerNorm in `models/layers`?

Reply:
When I use the models/layers implementation, I get an error because the model uses weights without bias in layernorm, and that implementation lacks a way to set just the bias to None.

Reply:
This is an updated version of the `model/layers`.

Reply:
However, I can add these changes to the layernorm implementation in `model/layers`.